app/decorators/models/solidus_social/spree/user_decorator.rb

```diff
@@ -12,9 +12,9 @@ def self.prepended(base)
       end
 
       def apply_omniauth(omniauth)
-        skip_signup_providers = SolidusSocial::OAUTH_PROVIDERS.map { |p| p[1] if p[2] == 'true' }.compact
-        if skip_signup_providers.include? omniauth['provider']
-          self.email = omniauth['info']['email'] if email.blank?
+        skip_signup_providers = SolidusSocial::OAUTH_PROVIDERS.map { |p| p.key if p.skip_signup == true }.compact
```

Comment:
`SolidusSocial::OAUTH_PROVIDERS.select { |provider| provider.skip_signup }.map(&:key)` is better for readability IMO.